dp_check/dp_check.go

```diff
@@ -142,32 +235,103 @@ func main() {
 	if syscall.Getuid() != 0 {
 		infoLog.Println("Not running as root, some checks may fail.")
 	}
-	// Check IPv6
-	runCheck("IPv6 routes", func() error {
-		var out string
-		var err error
-		cmd := "ip -6 route show"
-		infoLog.Printf("Check IPv6 routes with:|%v|...\n", cmd)
-		if out, err = getOutput(cmd); err != nil {
+
+	// Check if dp_check is running on a VM
+	runCheck("Running on GCE VM", func() error {
+		ret, err := isRunningOnGCP()
+		if err != nil {
 			return err
 		}
-		if !strings.Contains(out, "2001:4860:8040::/42") {
-			return fmt.Errorf(`Missing route prefix to backends: 2001:4860:8040::/42.
-IPv6 route setup either failed or hasn't been attempted`)
+		if !ret {
+			return fmt.Errorf("dp_check is not running on a GCE VM, this tool will not work as intended")
 		}
 		return nil
 	})
-	runCheck("IPv6 addresses", func() error {
-		var out string
-		var err error
-		cmd := "ip -6 addr show"
-		infoLog.Printf("Check IPv6 addresses with:|%v|...\n", cmd)
-		if out, err = getOutput(cmd); err != nil {
+
+	// Check connection to metadata server
+	runCheck("DirectPath availability", func() error {
+		client := &http.Client{}
+		metadataServerUrl := "http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ipv6s"
+		req, err := http.NewRequest("GET", metadataServerUrl, nil)
+		if err != nil {
+			return err
+		}
+		req.Header.Add("Metadata-Flavor", "Google")
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
 			return err
 		}
-		if !strings.Contains(out, "inet6 2600") {
+		ipv6FromMetadataServer = net.ParseIP(strings.TrimSuffix(string(body), "\n"))
+
+		infoLog.Printf("Check if this VM enables DirectPath by sending http GET request to %s", metadataServerUrl)
```

Comment:
nit: please put this log message at the top of this `runCheck` function, so that it's unconditionally logged before it's attempted